File: backend/app/schemas/importacion.py
from typing import Literal
from pydantic import BaseModel


class FilaImportacion(BaseModel):
    """Una fila tal como viene de la planilla — todo texto crudo, sin
    interpretar todavía. El mismo shape se usa para mandar la vista previa
    al frontend y para que el frontend la devuelva sin tocar al confirmar."""
    nombre: str = ""
    tipo: str = ""
    telefono: str = ""
    email: str = ""
    direccion: str = ""
    localidad: str = ""
    notas: str = ""
    servicio: str = ""
    frecuencia: str = ""
    precio_acordado: str = ""


class FilaImportacionPreview(BaseModel):
    fila: int
    datos: FilaImportacion
    estado: Literal["ok", "advertencia", "error", "duplicado"]
    mensajes: list[str]
    # Campos solo para mostrar en la vista previa (no se vuelven a mandar al confirmar).
    tipo_resuelto: str
    es_fijo: bool
    servicio_resuelto: str | None = None


class ResumenImportacion(BaseModel):
    creados: int
    contratos_creados: int
    omitidos: int
    detalles: list[FilaImportacionPreview]

File: frontend/src/api/importacion.ts
import { api } from "./client";

export interface FilaImportacionDatos {
  nombre: string;
  tipo: string;
  telefono: string;
  email: string;
  direccion: string;
  localidad: string;
  notas: string;
  servicio: string;
  frecuencia: string;
  precio_acordado: string;
}

export type EstadoFilaImportacion = "ok" | "advertencia" | "error" | "duplicado";

export interface FilaImportacionPreview {
  fila: number;
  datos: FilaImportacionDatos;
  estado: EstadoFilaImportacion;
  mensajes: string[];
  tipo_resuelto: string;
  es_fijo: boolean;
  servicio_resuelto: string | null;
}

export interface ResumenImportacion {
  creados: number;
  contratos_creados: number;
  omitidos: number;
  detalles: FilaImportacionPreview[];
}

export const importacionApi = {
  plantillaUrl: () => `${api.defaults.baseURL}/clientes/importar/plantilla`,

  previsualizar: (archivo: File) => {
    const formData = new FormData();
    formData.append("archivo", archivo);
    return api
      .post<FilaImportacionPreview[]>("/clientes/importar/previsualizar", formData, {
        headers: { "Content-Type": "multipart/form-data" },
      })
      .then((r) => r.data);
  },

  confirmar: (filas: FilaImportacionDatos[]) =>
    api.post<ResumenImportacion>("/clientes/importar/confirmar", filas).then((r) => r.data),
};
